tests/search/containers/TestCurrentSearchPage.js

```diff
@@ -1,17 +1,64 @@
-import { mapStateToProps } from 'search/containers/CurrentSearchPage';
+import * as actions from 'search/actions';
+import { mapDispatchToProps, mapStateToProps } from 'search/containers/CurrentSearchPage';
+import * as api from 'core/api';
 
 describe('CurrentSearchPage.mapStateToProps', () => {
-  const props = mapStateToProps({
-    addons: {ab: {slug: 'ab', title: 'ad-block'},
-             cd: {slug: 'cd', title: 'cd-block'}},
-    search: {query: 'ad-block'},
+  const state = {
+    addons: {ab: {slug: 'ab', name: 'ad-block'},
+             cd: {slug: 'cd', name: 'cd-block'}},
+    search: {query: 'ad-block', loading: false, results: [{slug: 'ab', name: 'ad-block'}]},
+  };
+  const props = mapStateToProps(state);
+
+  it('passes the search state', () => {
+    assert.strictEqual(props, state.search);
+  });
+});
+
+describe('CurrentSearchPage.mapDispatchToProps', () => {
+  let dispatch;
+  let mocks;
+
+  beforeEach(() => {
+    dispatch = sinon.spy();
+    mocks = {
+      actions: sinon.mock(actions),
+      api: sinon.mock(api),
+    };
+  });
+
+  afterEach(() => {
+    Object.keys(mocks).forEach((name) => mocks[name].restore());
   });
 
-  it('passes the query through', () => {
-    assert.equal(props.query, 'ad-block');
+  function handleSearch(query, response = Promise.resolve({})) {
+    mocks.api.expects('search').withArgs({ query }).once().returns(response);
+    return mapDispatchToProps(dispatch).handleSearch(query);
+  }
+
+  it('sets the query', (done) => {
+    const searchAction = sinon.stub();
+    mocks.actions.expects('searchStart').withArgs('DuckDuckGo').once().returns(searchAction);
+    handleSearch('DuckDuckGo').then(() => done()).catch(done);
+    assert(dispatch.calledWith(searchAction), 'expected action to be dispatched');
+    mocks.actions.verify();
   });
 
-  it('filters the add-ons', () => {
-    assert.deepEqual(props.results, [{slug: 'ab', title: 'ad-block'}]);
+  it('sets the response', (done) => {
+    const loadAction = sinon.stub();
+    const response = Promise.resolve({
+      entities: {addons: {foo: {}, bar: {}}},
+      result: ['foo', 'bar'],
+    });
+    mocks.actions
+      .expects('searchLoad')
+      .withArgs({query: 'Yahoo!', entities: {addons: {foo: {}, bar: {}}}, result: ['foo', 'bar']})
+      .once()
+      .returns(loadAction);
+    handleSearch('Yahoo!', response).then(() => {
```

Comment:
Since this returns a promise if you just simply return it you won't need to use `done` and failure should get handled.